akka-stream/src/main/scala/akka/stream/scaladsl/Flow.scala

```diff
@@ -2572,6 +2601,17 @@ trait FlowOpsMat[+Out, +Mat] extends FlowOps[Out, Mat] {
   def divertToMat[Mat2, Mat3](that: Graph[SinkShape[Out], Mat2], when: Out ⇒ Boolean)(matF: (Mat, Mat2) ⇒ Mat3): ReprMat[Out, Mat3] =
     viaMat(divertToGraph(that, when))(matF)
 
+  /**
+   * Creates a real `Flow` upon receiving the first element. Internal `Flow` will not be created
+   * if there are no elements, because of completion or error. Then transform this `Flow`
+   * with the created `Flow`. The materialized value of the combined `Flow` will be
+   * calculated in the same way as in [[Flow#viaMat viaMat]].
+   *
+   * @see [[Flow#viaLazyInit]]
+   */
+  def viaLazyInitMat[T, Mat2, Mat3](flowFactory: Out ⇒ Future[Flow[Out, T, Mat2]], fallback: () ⇒ Mat2)(combine: (Mat, Mat2) ⇒ Mat3): ReprMat[T, Mat3] =
+    viaMat(Flow.fromGraph(new LazyFlow(flowFactory, fallback)))(combine)
+
```

Comment:
I'd say those are optional, though perhaps ok to have... the main thing to add would be `Flow.lazyInit` as in, on the companion object?

Reply:
Yeah, probably better idea - and remove those two, so everyone can choose whether they want `via` or `viaMat` or whatever have you. Thanks.

Reply:
I agree